api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -52,17 +57,50 @@ public Obs getObsByUuid(String uuid) {
 	
 	@Override
 	public Collection<Obs> searchForObservations(ReferenceParam encounterReference, ReferenceParam patientReference,
+	        ReferenceParam hasMemberReference, TokenAndListParam valueConcept, DateRangeParam valueDateParam,
+	        QuantityAndListParam valueQuantityParam, StringAndListParam valueStringParam, DateRangeParam date,
 	        TokenAndListParam code, SortSpec sort) {
 		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Obs.class);
 		
 		handleEncounterReference("e", encounterReference).ifPresent(c -> criteria.createAlias("encounter", "e").add(c));
 		handlePatientReference(criteria, patientReference, "person");
+		handleHasMemberReference("hm", hasMemberReference).ifPresent(c -> criteria.createAlias("groupMembers", "hm").add(c));
+		handleValueCodedConcept(criteria, valueConcept);
+		handleDateRange("valueDatetime", valueDateParam);
+		
+		handleValueStringParam("valueText", valueStringParam).ifPresent(criteria::add);
+		handleValueQuantityParam("valueNumeric", valueQuantityParam).ifPresent(criteria::add);
+		handleDateRange("obsDatetime", date);
 		handleCodedConcept(criteria, code);
 		handleSort(criteria, sort);
 		
 		return criteria.list();
 	}
 	
+	protected Optional<Criterion> handleValueQuantityParam(@NotNull String propertyName,
+	        QuantityAndListParam valueQuantityParam) {
+		if (valueQuantityParam == null) {
+			return Optional.empty();
+		}
+		return handleAndListParam(valueQuantityParam,
+		    v -> Optional.of(ilike(propertyName, v.getValueAsString(), MatchMode.EXACT)));
```

Comment:
We shouldn't use `ilike` here, which is only for matching strings. We should use `eq`.

Reply:
@ibacher Ok I will change this. Is everything else fine, should I complete the code & tests and update the PR?

Reply:
I haven't really gone through everything else here, but please do keep this up to date